src/runtime/FunctionObjectInlines.h

```diff
@@ -91,13 +91,7 @@ class FunctionObjectProcessCallGenerator {
     template <typename FunctionObjectType, bool isConstructCall, bool hasNewTargetOnEnvironment, bool canBindThisValueOnEnvironment, typename ThisValueBinder, typename NewTargetBinder, typename ReturnValueBinder>
     static ALWAYS_INLINE Value processCall(ExecutionState& state, FunctionObjectType* self, const Value& thisArgument, const size_t argc, Value* argv, Object* newTarget) // newTarget is null on [[call]]
     {
-#ifdef STACK_GROWS_DOWN
-        if (UNLIKELY(state.stackLimit() > (size_t)currentStackPointer())) {
-#else
-        if (UNLIKELY(state.stackLimit() < (size_t)currentStackPointer())) {
-#endif
-            ErrorObject::throwBuiltinError(state, ErrorCode::RangeError, "Maximum call stack size exceeded");
-        }
+        state.checkStackOverflow();
```

Comment:
IMO calling non-inline checkStackOverflow function affect performance.
because here is hot spot
I think we can divide checkStackOverflow into two function check overflow and throw exception for inlining

Reply:
Thanks for the review!
I fixed it to embed the check code instead of function calls